src/Compilers/CSharp/Portable/Syntax/SyntaxFacts.cs

```diff
@@ -244,7 +244,7 @@ public static bool IsInNamespaceOrTypeContext(ExpressionSyntax? node)
                     switch (parent.Kind())
                     {
                         case UsingDirective:
-                            return ((UsingDirectiveSyntax)parent).Name == node;
+                            return ((UsingDirectiveSyntax)parent).Type == node;
```

Comment:
What if `node` is part of the `Type`, but not equal to the `Type`?

Reply:
Same login as before.   Previous code never allowed for a nested type to be considered here.

Reply:
Changed this.  This should now fallout from teh change to IsTypeContext.

Reply:
The new logic is correct.  The 'type' of a `using` using directive is a type-or-namespace context.